AppController/djinn.rb

```diff
@@ -4927,6 +4855,40 @@ def find_lowest_free_port(starting_port)
   end
 
 
+  # Updates @app_info_map with registered instances.
```

Comment:
can you add a note about how this must be called under APPS_LOCK

it currently is from scale_deployment since that is only ever run under APPS_LOCK, so there's no problem.